File: app/src/main/java/com/earthdefensesystem/androidanimatedimages/MainActivity.java
package com.earthdefensesystem.androidanimatedimages;

import android.content.Intent;
import android.graphics.drawable.Animatable;
import android.graphics.drawable.Drawable;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.ImageView;

public class MainActivity extends AppCompatActivity {
    private ImageView animatedButton;
    private Button nextActivity, gifActivity;
    private boolean isClicked = true;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        animatedButton = findViewById(R.id.image_button);
        nextActivity = findViewById(R.id.button);
        gifActivity = findViewById(R.id.button2);


        animatedButton.setImageDrawable(getDrawable(R.drawable.switch_vector_1));
        final Drawable drawable = (animatedButton).getDrawable();
        if (drawable instanceof Animatable) {
            ((Animatable) drawable).start();
        }

        nextActivity.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Intent i = new Intent(getApplicationContext(),ListVector.class);
                startActivity(i);
            }
        });
        gifActivity.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Intent i = new Intent(getApplicationContext(),GifActivity.class);
                startActivity(i);
            }
        });

        animatedButton.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                if (isClicked) {
                    animatedButton.setImageDrawable(getDrawable(R.drawable.switch_vector_2));
                    isClicked = false;
                } else {
                    animatedButton.setImageDrawable(getDrawable(R.drawable.switch_vector_1));
                    isClicked = true;
                }
                final Drawable drawable = animatedButton.getDrawable();
                if (drawable instanceof Animatable) {
                    ((Animatable) drawable).start();
                }
            }
        });
    }
}

File: app/src/main/java/com/example/androidanimatedimages/MainActivity.kt
package com.example.androidanimatedimages

import android.annotation.SuppressLint
import android.graphics.ImageDecoder
import android.graphics.drawable.Animatable
import android.graphics.drawable.AnimatedImageDrawable
import android.graphics.drawable.AnimationDrawable
import android.os.Build
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.widget.ImageView
import androidx.core.content.ContextCompat
import kotlinx.android.synthetic.main.activity_main.*

class MainActivity : AppCompatActivity() {

    val drawablesIds = listOf(R.drawable.edgif, R.drawable.warhammer_animated)
    var pointer = 0
    fun incrementPointer(){
        pointer++
        if (pointer < 0){
            pointer = drawablesIds.size - 1 
        }
    }
       fun decrementPointer() {
           pointer--
           if (pointer < 0) {
               pointer = drawablesIds.size - 1
           }
       }

    @SuppressLint("WrongThread")
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)


        button_next.setOnClickListener {
            incrementPointer()
            animated_image.setImageDrawable(ContextCompat.getDrawable(this, drawablesIds[pointer]))

        }

        button_previous.setOnClickListener {
            decrementPointer()
            animated_image.setImageDrawable(ContextCompat.getDrawable(this,drawablesIds[pointer]))
        }


        button_play.setOnClickListener {view ->
            when(pointer){
                0 -> animateGif(drawablesIds[pointer], animated_image)
                1 -> animateAnimationDrawable(drawablesIds[pointer], animated_image)
            }
            animateVectorDraw(R.drawable.avd_anim,  view as ImageView)
        }

        //animateGif()
        //animateAnimationDrawable()

       // animateVectorDraw()

    }

    private fun animateVectorDraw(id: Int, view: ImageView) {
        val animatedVectorDrawable = ContextCompat.getDrawable(this, id)
        view.setImageDrawable(animatedVectorDrawable)
        (animatedVectorDrawable as Animatable).start()
    }

    private fun animateAnimationDrawable(id: Int, view: ImageView) {
        val frameDrawable = ContextCompat.getDrawable(this,id)
        view.setImageDrawable(frameDrawable)
        (frameDrawable as AnimationDrawable).start()
    }

    private fun animateGif(id: Int, view: ImageView) {
        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.P) {
            val gifDrawable = ImageDecoder.decodeDrawable(
                ImageDecoder.createSource(resources, id))
            view.setImageDrawable(gifDrawable)
            (gifDrawable as AnimatedImageDrawable).start()
        }
    }
}
